gammapy/modeling/models/spectral.py

```diff
@@ -1094,6 +1096,7 @@ def __init__(
     def evaluate(energy, amplitude, reference, expfactor, index_1, index_2):
         """Evaluate the model (static function)."""
         pwl = amplitude * (energy / reference) ** (-index_1)
+        expfactor = expfactor / reference.unit ** index_2
```

Comment:
I think this line should move in the `try` block, otherwise it will fail for the evaluation with `ufloat`.